organisaatio-ui/src/components/Sivut/LomakeSivu/Koulutustoimija/OrganisaatioHistoriaLomake/OrganisaatioHistoriaLomake.tsx

```diff
@@ -1,118 +1,136 @@
 import * as React from 'react';
+import { useContext } from 'react';
 import styles from './OrganisaaatioHistoriaLomake.module.css';
 import YksinkertainenTaulukko from '../../../../Taulukot/YksinkertainenTaulukko';
 import useAxios from 'axios-hooks';
 import Spin from '@opetushallitus/virkailija-ui-components/Spin';
+import { LanguageContext } from '../../../../../contexts/contexts';
+import { Link } from 'react-router-dom';
 
 type organisaatioHistoriaProps = {
     oid?: string;
 };
 
 export default function OrganisaatioHistoriaLomake(props: organisaatioHistoriaProps) {
+    const { i18n, language } = useContext(LanguageContext);
     const [{ data: historia, loading: historiaLoading, error: historiaError }] = useAxios(
-        `/organisaatio/organisaatio/v4/1.2.246.562.10.69981965515/historia`
+        `/organisaatio/organisaatio/v4/${props.oid}/historia`
     );
-    console.log('hist', historia);
+    console.log('hist', historiaLoading, historia);
+    if (!historia || historiaLoading || historiaError) {
+        return <Spin />;
+    }
+
+    const renderLink = (a) => {
+        return (
+            <Link to={`/lomake/${a.oid}`}>
+                {(a.nimi[language] || a.nimi['fi'] || a.nimi['sv'] || a.nimi['en']) +
+                    (a.status !== 'AKTIIVINEN' ? '(' + i18n.translate('LABEL_' + a.status.toUpperCase()) + ')' : '')}
+            </Link>
+        );
+    };
+    const historiaSorter = (a, b) => {
+        return a.alkuPvm.localeCompare(b.alkuPvm);
+    };
+    const historiaMapper = (a, key) => {
+        return {
+            oid: a[key].oid,
+            nimiHref: renderLink(a[key]),
+            alkuPvm: a.alkuPvm,
+            loppuPvm: a.loppuPvm,
+            status: a[key].status,
+        };
+    };
+
     const liittyneetColumns = [
         {
-            Header: 'Nimen voimassaolo',
+            Header: i18n.translate('RAKENNE_LIITOSPVM'),
             accessor: 'alkuPvm',
         },
         {
-            Header: 'Liittyneet organisaatiot',
-            accessor: 'organisaatiot',
+            Header: i18n.translate('RAKENNE_LIITTYNEET_NIMI'),
+            accessor: 'nimiHref',
         },
     ];
-    const liittettyYlempaanColumns = [
+    const yhdistettyColumns = [
         {
-            Header: 'Alkupäivämäärä',
+            Header: i18n.translate('RAKENNE_LIITOSPVM'),
             accessor: 'alkuPvm',
         },
         {
-            Header: 'Loppupäivämäärä',
-            accessor: 'loppuPvm',
-        },
-        {
-            Header: 'Ylemmän tason organisaatio',
-            accessor: 'organisaatiot',
+            Header: i18n.translate('RAKENNE_YHDISTETTY_NIMI'),
+            accessor: 'nimiHref',
         },
     ];
-    const liittettyYlempaanData = [
+    const ylemmanTasonColumns = [
```

Comment:
might not make sense but seems like column creation could be parameterized based on two input strings? `[[translationKey, accessor],...].map(column => ({ Header: i18n.translate(translationKey), accessor}))`